libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/new.size_align.except.pass.cpp

```diff
@@ -9,6 +9,10 @@
 // UNSUPPORTED: no-exceptions
 // UNSUPPORTED: sanitizer-new-delete
 
+// Libc++ when built for z/OS doesn't contain the aligned allocation functions,
+// nor does the dynamic library shipped with z/OS.
+// UNSUPPORTED: target={{.+}}-zos{{.*}}
```

Comment:
I assume at some point in the future z/OS will support this feature. So maybe make this `XFAIL` instead of `UNSUPPORTED`. That way there is a notification when support is available.

Reply:
Thanks for the review! In the past, we marked similar testcases with UNSUPPORTED, so I would like them all to be consistent. So should I change those all to XFAIL as well?

Reply:
I agree with consistency, both for z/OS and our generic way. Grepping for `XFAIL:.*target=` has 101 hits, `UNSUPPORTED:.*target=` 44 hits where 25 are due to zos. 
Could you make a separate review for the existing `UNSUPPORTED` -> `XFAIL` changes? Feel free to assign that patch to me. If you can assign, please leave a link in this review.

Reply:
Sounds good! I've made the changes to this PR and created and assigned a PR to you to fix the other 44 here https://github.com/llvm/llvm-project/pull/81513